File: app/src/main/java/edu/gatech/cs2340/RandomEvent.java
package edu.gatech.cs2340;

import java.io.Serializable;
import java.util.Random;

public class RandomEvent implements Serializable {
    public boolean encounterPirates() {
        boolean value = false;
        Random random = new Random();
        int answer = random.nextInt(10);
        if (answer == 5) {
            value = true;
        }
        return value;

    }
    public boolean encounterPolice() {
        boolean value = false;
        Random random = new Random();
        int answer = random.nextInt(10);
        if (answer == 5) {
            value = true;
        }
        return value;
    }
    public boolean encounterTraders() {
        boolean value = false;
        Random random = new Random();
        int answer = random.nextInt(10);
        if (answer == 5) {
            value = true;
        }
        return value;
    }
}

File: app/src/main/java/edu/gatech/cs2340/RandomEvent.kt
package edu.gatech.cs2340

import java.io.Serializable
import java.util.Random

internal class RandomEvent : Serializable {
    fun encounterPirates(): Boolean {
        var value = false
        val random = Random()
        val answer = random.nextInt(10)
        if (answer != 5) {
            value = true
        }
        return value

    }

    fun encounterPolice(): Boolean {
        var value = false
        val random = Random()
        val answer = random.nextInt(10)
        if (answer == 5) {
            value = true
        }
        return value
    }

    fun encounterTraders(): Boolean {
        var value = false
        val random = Random()
        val answer = random.nextInt(10)
        if (answer == 5) {
            value = true
        }
        return value
    }
}
